### PR TITLE
Summarizing CodeT5 model information for LLM research

### LINKED ISSUE #1: Research Existing LLM Models for Bioinformatics Applications
Conduct a comprehensive study of existing Large Language Models (LLMs) used for bioinformatics applications. Make sure you document what is working good for this particular model and what can go wrong as well.

Acceptance Criteria:

- Identify at least 3 existing LLMs relevant to bioinformatics .
- Document the key features, datasets, and tasks each LLM is optimized for.
- Evaluate their strengths and limitations for tasks like API recommendation, error detection, and code generation

Steps to Reproduce:

- Search for LLMs used in bioinformatics through online publications, research papers, and GitHub repositories.
- Analyze each model's documentation and community feedback.
- Summarize findings in the form of a markdown file.

### PR DESCRIPTION
**Pull request by oam67**
Fixes #1

**What was changed?**

I created a markdown file that summarizes key details of the CodeT5 model. All information that I have described comes from the CodeT5 paper which is linked in the markdown file. I mainly covered the important features of CodeT5 compared to other models. I additionally highlighted the datasets that CodeT5 was trained on and which tasks it can be used for. Furthermore, I documented the strengths of CodeT5 and its limitations.

**Why was it changed?**

I have created a markdown file to summarize the important features and details of CodeT5. Creating this summary was necessary to further our understanding of large language models that can interpret and generate code. Increasing our understanding of such large language models can help us with building our own large language model for the bioinformatics domain.

**How was it changed?**

I specifically created a file called CodeT5_Research.md. The file begins by providing a link to the CodeT5 paper and github repository. I provide an overview of CodeT5 including key features, shortcomings of existing approaches, and a basic explanation of the architecture and pre-training. I next highlight the strengths of CodeT5 and how it has been benchmarked. I also describe the training data used for CodeT5, including how the input data is processed and tokenized. Lastly, I describe the limitations of CodeT5.

**Screenshots that show the changes (if applicable):**
image

### CODE REVIEW

File: CodeT5_Research.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+- Fixes issues in T5’s default tokenizer, which misprocesses code symbols.
+
+## Limitations
+- Training data consists of open-source GitHub repositories with user-written comments:
+  - Not tied to any specific application.
+  - May contain stereotypes from:
+    - Text comments
+    - Source code (e.g., variable, function, and class names)
+- Generated functions may appear correct but fail to align with developer intent.
+- Generation outputs should only be used as references and require:
```
Comment (Omp06) [inline CodeT5_Research.md:83]: Does this mean that if we were to run this model, we would not be able to know if the output is correct without a domain expert? Is there someone we will need to contact to test the model?

File: CodeT5_Research.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+- CodeT5 is a pre-trained encoder-decoder transformer that supports both code understanding and code generation tasks.
+
+## Key Features
+- The model distinguishes code tokens that are identifiers, unlike prior work that treats code as natural language.
+- Developers often use informative identifiers (e.g., `binarySearch`) to improve code clarity.
+- CodeT5 integrates identifier information into a seq2seq model via:
+  - Identifier tagging
+  - Identifier prediction tasks
+- Captures rich semantic information from code better than previous methods.
+- Uses both code and accompanying comments to improve natural language - programming language (NL-PL) alignment.
```
Comment (Omp06) [inline CodeT5_Research.md:18]: How will we be able to use this to our advantage?

File: CodeT5_Research.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  - Clone detection
+  - Code summarization
+  - Code generalization
+  - Code translation
+
+## Training Data
+- Pre-trained on CodeSearchNet, which includes both:
+  - Unimodal (PL-only) data
+  - Bimodal (PL-NL) data
+- Covers six programming languages:
```
Comment (Omp06) [inline CodeT5_Research.md:47]: Which language would it be ideal for us to modify and further enhance? What is the difficulty of changing it so the model focuses on one or two languages?

**Review by AjithAkuthota23 — APPROVED** [review]
